tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -794,12 +794,21 @@ def _make_matrix(self, dtype, xp):
         return a, b
 
     def _test_spsolve_triangular(self, sp, a, b):
-        return sp.linalg.spsolve_triangular(a, b, lower=self.lower,
-                                            unit_diagonal=self.unit_diagonal)
+        def call_with_ignore_warning(f):
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', sp.SparseEfficiencyWarning)
+                return f()
+        if sp is scipy.sparse:
+            call = call_with_ignore_warning
+        else:
+            call = lambda f: f()
+        return call(lambda: sp.linalg.spsolve_triangular(
+            a, b, lower=self.lower, unit_diagonal=self.unit_diagonal))
 
     @pytest.mark.parametrize('format', ['csr', 'csc', 'coo'])
     @testing.for_dtypes('fdFD')
-    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp')
+    @testing.numpy_cupy_allclose(
+        rtol=1e-5, atol=1e-5, sp_name='sp', contiguous_check=False)
```

Comment:
Add `contiguous_check=False` as spsm assumes f-order for the solution matrix `x` (on the other hand, csrsm2 overwrites b so the solution matrix has the same order with b)